apps/atomy-q/API/app/Adapters/InsightOperations/ProviderInsightNarrativeAdapter.php

```
@@ -0,0 +1,108 @@
+use Nexus\InsightOperations\DTOs\AiArtifactProvenanceDto;
+use Nexus\IntelligenceOperations\DTOs\AiStatusSchema;
+
+final readonly class ProviderInsightNarrativeAdapter implements InsightNarrativePortInterface
+{
+    public function __construct(
+        private ProviderInsightClientInterface $client,
+        private ClockInterface $clock,
+    ) {
+    }
+
+    public function generate(string $featureKey, string $tenantId, string $subjectType, string $actorId, array $facts): AiArtifactDto
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$actorId'. (undefined)

(UnusedFormalParameter)